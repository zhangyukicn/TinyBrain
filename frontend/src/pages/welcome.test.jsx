
import React from 'react';
import { render, screen } from '@testing-library/react';
import { configure, EnzymeAdapter, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Lobby from './lobby';
import Welcome from './welcome';
configure({ adapter: new Adapter() });

// Welcome page
describe('Welcome', () => {
    const noop = () => { }
    const modalStarta = shallow(<Welcome />);

    it('is matched with snapshot', () => {
      expect(modalStarta).toMatchSnapshot();
    });

    it('triggers Login onClike event handler when clicked', () => {
      const onClick = jest.fn();
      shallow(<goToLogIn onClick={onClick} />).simulate('click');
      expect(onClick).toHaveBeenCalledTimes(1);
    })

    it('triggers playgame onClike event handler when clicked', () => {
      const onClick = jest.fn();
      shallow(<goToPlay onClick={onClick} />).simulate('click');
      expect(onClick).toHaveBeenCalledTimes(1);
    })
  }
  )
