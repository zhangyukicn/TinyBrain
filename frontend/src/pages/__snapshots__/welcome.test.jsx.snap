// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome is matched with snapshot 1`] = `ShallowWrapper {}`;
